src/cli/index.js

```diff
@@ -98,90 +94,55 @@ commander.option('--network-concurrency <number>', 'maximum number of concurrent
 commander.option('--network-timeout <milliseconds>', 'TCP timeout for network requests', parseInt);
 commander.option('--non-interactive', 'do not show interactive prompts');
 
-// get command name
-let commandName: ?string = args.shift() || '';
-let command;
-
-//
 const getDocsLink = (name) => `${constants.YARN_DOCS}${name || ''}`;
 const getDocsInfo = (name) => 'Visit ' + chalk.bold(getDocsLink(name)) + ' for documentation about this command.';
 
-//
+// get command name
+let commandName: string = args.shift() || 'install';
+
 if (commandName === '--help' || commandName === '-h') {
   commandName = 'help';
 }
 
-// if no args or command name looks like a flag then default to `install`
-if (!commandName || commandName[0] === '-') {
-  if (commandName) {
-    args.unshift(commandName);
-  }
-  commandName = 'install';
+if (args.indexOf('--help') >= 0 || args.indexOf('-h') >= 0) {
+  args.unshift(commandName);
+  commandName = 'help';
 }
 
-// aliases: i -> install
-if (commandName && typeof aliases[commandName] === 'string') {
-  const alias = aliases[commandName];
-  command = {
-    run(config: Config, reporter: ConsoleReporter | JSONReporter): Promise<void> {
-      throw new MessageError(`Did you mean \`yarn ${alias}\`?`);
-    },
-  };
+// if no args or command name looks like a flag then set default to `install`
+if (commandName[0] === '-') {
+  args.unshift(commandName);
+  commandName = 'install';
 }
 
-//
-if (commandName === 'help' && args.length) {
-  commandName = camelCase(args.shift());
-  args.push('--help');
+let command;
+const camelised = camelCase(commandName);
+if (camelised) {
+  command = commands[camelised];
```

Comment:
I added a `|| ''` because `camelCase` function return null in some cases.

The final version:

```
commandName = camelCase(commandName) || '';

if (!Object.prototype.hasOwnProperty.call(commands, commandName)) {
    args.unshift(commandName);
    commandName = 'run';
}

command = commands[commandName];
```